src/mame/drivers/viper.cpp

```diff
@@ -2472,6 +2494,23 @@ void viper_state::init_vipercf()
 	m_maincpu->space(AS_PROGRAM).install_readwrite_handler(0xff300000, 0xff300fff, read64s_delegate(*this, FUNC(viper_state::unk_serial_r)), write64s_delegate(*this, FUNC(viper_state::unk_serial_w)));
 }
 
+void viper_state::init_viperppp()
+{
+	init_viperhd();
+	m_maincpu->space(AS_PROGRAM).install_read_handler(0xff400200, 0xff40023f, read16sm_delegate(*this, FUNC(viper_state::ppp_sensor_r)));
```

Comment:
Please use a derived memory map and machine configuration rather than adding init functions to install handlers.  Really the other ones should be done with memory maps as well.

Reply:
Agreed that the others should also really be memory maps, but I thought maybe it would be better to keep with the rest of the driver in this case. Maybe someone will give this driver some love some day and rework the rest of the map stuff. I turned the pp2nd part into a machine configuration at least.